src/components/sidebar/lineView/LineInfoTab.tsx

```diff
@@ -0,0 +1,213 @@
+import ISearchLine from '~/models/searchModels/ISearchLine';
+import { IValidationResult } from '~/validation/FormValidator';
+import { TransitToggleButtonBar, Dropdown } from '~/components/controls';
+import InputContainer from '../InputContainer';
+import * as s from './lineInfoTab.scss';
+
+interface ILineInfoTabState {
+    isLoading: boolean;
+}
+
+interface ILineInfoTabProps {
+    lineStore?: LineStore;
+    codeListStore?: CodeListStore;
+    errorStore?: ErrorStore;
+    isEditingDisabled: boolean;
+    isNewLine: boolean;
+    onChangeLineProperty: (property: string) => (value: any) => void;
+    invalidPropertiesMap: object;
+    setValidatorResult: (property: string, validationResult: IValidationResult) => void;
+}
+
+@inject('lineStore', 'codeListStore', 'errorStore')
+@observer
+class LineInfoTab extends React.Component<ILineInfoTabProps, ILineInfoTabState>{
+    private existingLines: ISearchLine[] = [];
+
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            isLoading: true,
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.isNewLine) {
+            this.fetchAllLines();
+        }
+    }
+
+    componentDidUpdate() {
+        if (this.props.isNewLine) {
+            this.fetchAllLines();
+        }
+    }
+
+    private selectTransitType = (transitType: TransitType) => {
+        this.props.onChangeLineProperty('transitType')(transitType);
+    }
+
+    private isLineAlreadyFound = (lineId: string): boolean => {
+        return Boolean(this.existingLines
+            .find((searchLine: ISearchLine) => searchLine.id === lineId));
+    }
+
+    private fetchAllLines = async () => {
+        if (this.existingLines.length > 0) return;
+
+        try {
+            const searchLines: ISearchLine[] = await LineService.fetchAllSearchLines();
+            this.existingLines = searchLines;
+        } catch (e) {
+            this.props.errorStore!.addError('Olemassa olevien linjojen haku ei onnistunut', e);
+        }
+    }
+
+    private onChangeLineId = (value: any) => {
+        this.props.onChangeLineProperty('id')(value);
+        if (this.isLineAlreadyFound(value)) {
+            const validationResult: IValidationResult = {
+                isValid: false,
+                errorMessage: `Linja ${value} on jo olemassa.`,
+            };
+            this.props.setValidatorResult('id', validationResult);
+        }
+    }
+
+    render() {
+        const line = this.props.lineStore!.line;
+        if (!line) return null;
+
+        const isEditingDisabled = this.props.isEditingDisabled;
+        const isUpdating = !this.props.isNewLine || this.props.isEditingDisabled;
+        const onChange = this.props.onChangeLineProperty;
+        const invalidPropertiesMap = this.props.invalidPropertiesMap;
+        const selectedTransitTypes = line!.transitType ? [line!.transitType!] : [];
+
+        return (
+        <div className={classnames(s.lineInfoTabView, s.form)}>
```

Comment:
Why not just set css properties of s.content inside s.lineInfoTabView, feels like we have unnecessary wrapping here